packages/common/http/src/transfer_cache.ts

```
@@ -110,7 +119,22 @@ export function transferCacheInterceptorFn(
   }
 
   const transferState = inject(TransferState);
-  const storeKey = makeCacheKey(req);
+
+  const originMap: Record<string, string> | null = inject(HTTP_TRANSFER_CACHE_ORIGIN_MAP, {
+    optional: true,
+  });
+  const isServer = isPlatformServer(inject(PLATFORM_ID));
+  if (ngDevMode && originMap && !isServer) {
+    console.warn(
```

We should probably throw an error here, looks for `throw new RuntimeError` elsewhere in to codebase this is how we do it.

---

Switched to a `RuntimeError`.
Since the interceptor function throws the error, a `RuntimeError` will be received when subscribing to the `HttpClient` instead of an `HttpErrorResponse`. That could be surprising. Are there other cases like that?

---

I know at least one https://github.com/angular/angular/blob/main/packages/common/http/src/xhr.ts#L66